components/script/dom/performanceobserver.rs

```diff
@@ -94,17 +95,29 @@ impl PerformanceObserver {
 impl PerformanceObserverMethods for PerformanceObserver {
     // https://w3c.github.io/performance-timeline/#dom-performanceobserver-observe()
     fn Observe(&self, options: &PerformanceObserverInit) -> Fallible<()> {
+        // step 1
         // Make sure the client is asking to observe events from allowed entry types.
         let entry_types = options.entryTypes.iter()
                                             .filter(|e| VALID_ENTRY_TYPES.contains(&e.as_ref()))
                                             .map(|e| e.clone())
                                             .collect::<Vec<DOMString>>();
+        // step 2
         // There must be at least one valid entry type.
         if entry_types.is_empty() {
             return Err((Error::Type("entryTypes cannot be empty".to_string())));
         }
 
         let performance = self.global().as_window().Performance();
+
+        // step 5
+        if options.buffered {
+            let pel = PerformanceEntryList::new(performance.GetEntries());
```

Comment:
Instead of getting the list of buffered entries here, could we move this block inside `add_observer` and pass the `buffered` flag along with the observer and the entry types, please?